docs/API-specification.md

```diff
@@ -0,0 +1,160 @@
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# API specification
+
+In order to be able to use yaml to define the objects required by apisix in k8s, the following structure is defined.
+
+## ApisixRoute
+
+`ApisixRoute` corresponds to the `route` object in apisix.
+
+Structure example:
+
+```yaml
+apiVersion: apisix.apache.org/v1
+kind: ApisixRoute 
+metadata:
+  annotations:                                          
+    k8s.apisix.apache.org/ingress.class: apisix_group
+    k8s.apisix.apache.org/ssl-redirect: 'false'
+  name: httpserverRoute
+  namespace: cloud
+spec:
+  rules:
+  - host: test.apisix.apache.org
+    http:
+      paths:
+      - backend:
+          serviceName: httpserver
+          servicePort: 8080
+        path: /hello*
+        plugins:
+          - httpserver-plugins
+          - ...      
+```
+
+|     Field     |  Type    | Description    |
+|---------------|----------|--------------------------------|
+| apiVersion    | string   | apisix.apache.org/v1           |
+| kind          | string   | ApisixRoute         |
+| name          | string   | httpserverRoute         |
+| namespace     | string   | Specify `namespace`, only one `backend` under `namespace` can be configured in the same yaml file. |
```

Comment:
You should, however, explain each fields in `spec`.

Reply:
updated.